en: add whats new in v1.3 (#1646)
pingcap/docs-tidb-operator: en/whats-new-in-v1.3.md

---
title: What's New in TiDB Operator 1.3
---

# What's New in TiDB Operator 1.3

TiDB Operator 1.3 introduces the following key features, which helps you manage TiDB clusters and the tools more easily in terms of extensibility, usability, and security.

## Compatibility changes

> - Enhance the feature of deploying a TiDB cluster across Kubernetes clusters. If you deployed a TiDB cluster across multiple Kubernetes clusters by TiDB Operator (<= v1.3.0-beta.1), upgrading TiDB Operator to v1.3.0 directly will cause failed rolling upgrade and the cluster might become abnormal. If you need to upgrade TiDB Operator from earlier versions to v1.3.0, take the following steps:

Reviewer: ```suggestion
- Enhance the feature of deploying a TiDB cluster across Kubernetes clusters. If you deploy a TiDB cluster across multiple Kubernetes clusters by TiDB Operator (<= v1.3.0-beta.1), upgrading TiDB Operator to v1.3.0 directly will cause failed rolling upgrade and the cluster might become abnormal. If you need to upgrade TiDB Operator from earlier versions to v1.3.0, take the following steps:
```

    1. Update CRD.

>     2. Add a new `spec.acrossK8s` field in TidbCluster spec and set it to `true`.

Reviewer: ```suggestion
    2. Add a new `spec.acrossK8s` field in the TidbCluster spec and set it to `true`.
```

## Rolling upgrade changes

- Optimize the default configuration of TiFlash (>= v5.4.0). Since v1.3.0-beta.1, TiDB Operator optimizes the default configuration for TiFlash >= v5.4.0. If you deploy a TiDB cluster >= v5.4 using TiDB Operator v1.2, after upgrading TiDB Operator to v1.3.x, the TiFlash component will be rolling updated. It is recommended to upgrade TiDB Operator to v1.3.x before upgrading the TiDB cluster to v5.4.x.

- TiKV under specific configurations might be rolling updated. If you deploy TiDB v5.0 or later versions, and configure `spec.tikv.seperateRocksDBLog: true` or `spec.tikv.separateRaftLog: true`, upgrading TiDB Operator to v1.3.x will cause TiKV to be rolling updated.

- To support configuring Prometheus shards in the TidbMonitor CR, upgrading TiDB Operator to v1.3.x will cause the TidbMonitor Pod to be rolling updated.

## Extensibility

- [Deploying a TiDB cluster across multiple Kubernetes clusters](deploy-tidb-cluster-across-multiple-kubernetes.md) becomes generally available. Now you can deploy [heterogeneous TiDB clusters](deploy-heterogeneous-tidb-cluster.md) across Kubernetes clusters.

> - Add a new `failover.recoverByUID` field to support one-time recover for TiKV/TiFlash/DM Worker.

Reviewer: ```suggestion
- Add a new `failover.recoverByUID` field to support one-time recovery for TiKV, TiFlash, and DM Worker.
```

- Support configuring [PodManagementPolicy](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#pod-management-policies) for the StatefulSets of TiDB cluster components.

> - Support configuring DNS config for Pods of all components.

Reviewer: ```suggestion
- Support configuring DNS for Pods of all components.
```

Author: 这里之前我也这样改了，研发说“配置 DNS 配置“和“配置 DNS”不一样😂
改成 “configuring DNS settings”？
<img width="791" alt="image" src="https://user-images.githubusercontent.com/59654584/154051108-d2f7d973-b68c-4589-8c22-dc9bde8af9fd.png">

Reviewer: 好的，可能研发想强调不是手动配的 DNS 而是通过配置文件配置的吗？
